lib/background_worker.rb

```diff
@@ -30,6 +30,12 @@ def self.verify_active_connections!
     end
   end
 
+  def self.release_connections!
+    if defined?(ActiveRecord) && ActiveRecord::VERSION::MAJOR == 4
+      ActiveRecord::Base.clear_all_connections!
```

Comment:
As the plan is to move towards Rails 5, and with Rails 5 being a reality, I think it would be a small change to include it in the condition. If the problem is verifying if the issue persists in Rails 5, can we at least add a story under the Rails Upgrade project?